plugins/woocommerce/src/Internal/DataStores/Orders/FeatureController.php

```diff
@@ -0,0 +1,324 @@
+<?php
+/**
+ * FeatureController class file.
+ */
+
+namespace Automattic\WooCommerce\Internal\DataStores\Orders;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * This is the main class that controls the custom orders table feature. Its responsibilities are:
+ *
+ * - Allowing to enable and disable the feature while it's in development (show_feature method)
+ * - Displaying UI components (entries in the tools page and in settings)
+ * - Providing the proper data store for orders via 'woocommerce_order_data_store' hook
+ *
+ * ...and in general, any functionality that doesn't imply database access.
+ */
+class FeatureController {
+
+	/**
+	 * The name of the option for enabling the usage of the custom orders table
+	 */
+	const CUSTOM_ORDERS_TABLE_USAGE_ENABLED_OPTION = 'woocommerce_custom_orders_table_enabled';
+
+	/**
+	 * The data store object to use.
+	 *
+	 * @var OrdersTableDataStore
+	 */
+	private $data_store;
+
+	/**
+	 * The data synchronizer object to use.
+	 *
+	 * @var DataSyncrhonizer
+	 */
+	private $data_synchronizer;
+
+	/**
+	 * Is the feature visible?
+	 *
+	 * @var bool
+	 */
+	private $is_feature_visible;
+
+	/**
+	 * Class constructor.
+	 */
+	public function __construct() {
+		$this->is_feature_visible = false;
+
+		$this->init_hooks();
+	}
+
+	/**
+	 * Initialize the hooks used by the class.
+	 */
+	private function init_hooks() {
+		add_filter(
+			'woocommerce_order_data_store',
+			function ( $default_data_store ) {
+				return $this->get_data_store_instance( $default_data_store );
+			},
+			10,
+			999
```

Comment:
The fourth argument to [`add_filter()`](https://developer.wordpress.org/reference/functions/add_filter/) is `$accepted_args`.

Reply:
Damn, I always put this one and the priority in the reverse order. Thanks!